### PR TITLE
CASMREL-1046 `kdump` overview

### PR DESCRIPTION
**Pull request by rustydb**
# Description

This provides a new section that describes `kdump`, its usage, and various test plans to use against it.

# Checklist Before Merging

- [x] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [x] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [x] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

File: background/ncn_kdump.md [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+> ***Q/A*** *Why does `kdump` look for the `ROOTRAID`?
+>
+> `kdump` looks for the `ROOTRAID` because the `/etc/fstab` file denotes that `/` is on `LABEL=ROOTRAID`. This entry in the `/etc/fstab` is
+> technically wrong for metal, but correct for Vagrant and Google Compute. The real root filesystem on metal is an overlayFS, but mounting the overlayFS
+> in `kdump` is complicated. To avoid complicating the `kdump` mount, `kdump` we provide `/` in the `/etc/fstab` file as `ROOTRAID`. The `ROOTRAID`
+> is the upper-directory of the overlayFS for the root filesystem, anything written here will appear as if it was written to a filesystem mounted at `/`. 
+
+### Enabling / disabling
+
+`kdump` is enabled when both the `kdump` package is installed *and* when the `crashkern` parameter is present on the command line.
```
Comment (jacobsalmela) [inline background/ncn_kdump.md:70]: `crashkernel`

File: background/ncn_kdump.md [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+    -rw-r--r-- 1 root root        192 Sep  7 14:33 README.txt
+    -rw-r--r-- 1 root root    4545158 Sep  7 14:33 System.map-5.3.18-150300.59.87-default
+    -rw------- 1 root root     139896 Sep  7 14:31 dmesg.txt
+    -rw------- 1 root root 2148950820 Sep  7 14:33 vmcore
+    -rw-r--r-- 1 root root   14157503 Sep  7 14:33 vmlinux-5.3.18-150300.59.87-default.gz   
+    ```
+
+## Analyzing a Dump
+
+A dump can be analzyed using the `crash` command. The analysis requires `kernel-default-debuginfo`
```
Comment (jacobsalmela) [inline background/ncn_kdump.md:365]: perhaps a note that the versions must match

**Review by jacobsalmela — CHANGES_REQUESTED** [review]

**Review by jacobsalmela — APPROVED** [review]

File: background/ncn_kdump.md [hunk 3]
```diff
@@ -0,0 +1,517 @@
...
+
+* The crash directory.
+* The kernel parameters used in the `kdump` `initrd`.
+
+### Dracut
+
+On metal artifacts, the `dracut-metal-mdsquash` module does some preparation for `kdump` by creating two symbolic links
+in the root overlay.
+
+* `$(lsblk -o MOUNTPOINT -nr /dev/disk/by-label/ROOTRAID)/boot` points to the actual `/boot` directory.
```
[the remark this answers is not in the document]
Comment (rustydb) [reply background/ncn_kdump.md:53]: Meant to prepend an `ls`, I want to provide the literal paths for folks to check out. The path is kind of dynamic, albeit it rarely changes.

File: background/ncn_kdump.md [hunk 4]
```diff
@@ -0,0 +1,517 @@
...
+
+`kdump` is configured by the `/etc/sysconfig/kdump` file, this file controls various aspects of `kdump`
+such as:
+
+* The crash directory.
+* The kernel parameters used in the `kdump` `initrd`.
+
+### Dracut
+
+On metal artifacts, the `dracut-metal-mdsquash` module does some preparation for `kdump` by creating two symbolic links
```
[the remark this answers is not in the document]
Comment (rustydb) [reply background/ncn_kdump.md:50]: Replacing with `squashFS`.

File: background/ncn_kdump.md [hunk 5]
```diff
@@ -0,0 +1,517 @@
...
+
+The dumps are conventionally written to `/var/crash` for analysis on the same machine following a reboot (assuming it
+doesn't crash again) or the disk can be relocated to a stable machine. If the dump is taken over the network, then analysis
+can be done using that network drive.
+
+For information on analyzing a dump, see [analyzing a dump](#analyzing-a-dump).
+
+## Usage
+
+This usage sections denotes how the non-compute nodes configure and set `kdump` up.
```
[the remark this answers is not in the document]
Comment (rustydb) [reply background/ncn_kdump.md:38]: This is how Minnesotan's speak.

File: background/ncn_kdump.md [hunk 6]
```diff
@@ -0,0 +1,517 @@
...
+
+These symbolic links are important for `kdump` to work. `kdump` will mount the `ROOTRAID` as the root filesystem, and then look for:
+
+* `/boot` to find the kernel image and `System.map` symbols file.
+* The crash directory specified in `/etc/sysconfig/kdump` (e.g. `/crash`)
+
+> ***Q/A*** *Why does `kdump` look for the `ROOTRAID`?
+>
+> `kdump` looks for the `ROOTRAID` because the `/etc/fstab` file denotes that `/` is on `LABEL=ROOTRAID`. This entry in the `/etc/fstab` is
+> technically wrong for metal, but correct for Vagrant and Google Compute. The real root filesystem on metal is an overlayFS, but mounting the overlayFS
```
[the remark this answers is not in the document]
Comment (rustydb) [reply background/ncn_kdump.md:64]: ur mom

File: background/ncn_kdump.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (heemstra) [reply background/ncn_kdump.md:53]: there's no closing paren

[the remark this answers is not in the document]
Comment (heemstra) [reply background/ncn_kdump.md:53]: oh. there is. not sure why I didn't see that the first time. but yeah, ls makes sense

File: background/ncn_kdump.md [hunk 7]
```diff
@@ -0,0 +1,517 @@
...
+> in `kdump` is complicated. To avoid complicating the `kdump` mount, `kdump` we provide `/` in the `/etc/fstab` file as `ROOTRAID`. The `ROOTRAID`
```
[the remark this answers is not in the document]
Comment (rustydb) [reply background/ncn_kdump.md]: I think I've fixed this, please recheck.

File: background/ncn_kdump.md [hunk 8]
```diff
@@ -0,0 +1,517 @@
...
+`kdump` is configured by the `/etc/sysconfig/kdump` file, this file controls various aspects of `kdump`
+such as:
+
+* The crash directory.
+* The kernel parameters used in the `kdump` `initrd`.
+
+### Dracut
+
+On metal artifacts, the `dracut-metal-mdsquash` module does some preparation for `kdump` by creating two symbolic links
+in the root overlay.
```
[the remark this answers is not in the document]
Comment (rustydb) [reply background/ncn_kdump.md]: Done.

File: background/ncn_kdump.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (rustydb) [reply background/ncn_kdump.md]: Fixed.

File: background/ncn_kdump.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (rustydb) [reply background/ncn_kdump.md]: Fixed.

**Review by heemstra — APPROVED** [review]

**Pull request merged**